picoquic/picoquic.h

```diff
@@ -449,13 +450,47 @@ int picoquic_prepare_packet(picoquic_cnx_t* cnx,
     uint64_t current_time, uint8_t* send_buffer, size_t send_buffer_max, size_t* send_length,
     struct sockaddr_storage * p_addr_to, int * to_len, struct sockaddr_storage * p_addr_from, int * from_len);
 
-/* send and receive data on streams */
+/* Mark stream as active, or not.
+ * If a stream is active, it will be polled for data when the transport
+ * is ready to send.
+ * Returns an error if data was previously queued on the stream using
+ * "picoquic_add_to_stream" and the data has not been sent yet.
+ */
+int picoquic_mark_active_stream(picoquic_cnx_t* cnx,
+    uint64_t stream_id, int is_active);
+
+/* If a stream is marked active, the application will receive a callback with
+ * event type "picoquic_callback_prepare_to_send" when the transport is ready to
+ * send data on a stream. The "length" argument in the call back indicates the
+ * largest amount of data that can be sent, and the "bytes" argument points
+ * to an opaque context structure. In order to prepare data, the application
+ * needs to call "picoquic_provide_stream_data_buffer" with that context
+ * pointer, with the number of bytes that it wants to write, with an indication
+ * of whether or not the fin of the stream was reached, and also an indication
+ * of whether or not the stream is still active. The function
+ * returns the pointer to a memory address where to write the byte -- or
+ * a NULL pointer in case of error. The application then copies the specified 
+ * number of bytes at the provided address, and provide a return code 0 from
+ * the callback in case of success, or non zero in case of error.
+ */
+
+uint8_t* picoquic_provide_stream_data_buffer(void* context, size_t nb_bytes, int is_fin, int is_still_active);
```

Comment:
Since there are only three possible values for (is_fin, is_still_active) tuple, I can see a case for an enum instead of two booleans.  But this is only a minor nit.

Reply:
I would rather not specify an enum.

In practice, it will most of the type be OK to just set "is_still_active=1". I am mostly concerned with thinks like video streaming, in which the application sends one frame, waits a bit, sends another frame, etc.